src/dbnode/integration/setup.go

```diff
@@ -465,7 +466,12 @@ func NewTestSetup(
 	}
 
 	for _, fn := range storageOptFns {
-		storageOpts = fn(storageOpts)
+		if fn != nil {
+			storageOpts = fn(storageOpts)
+			if storageOpts != nil && storageOpts.AdminClient() == nil {
+				storageOpts = storageOpts.SetAdminClient(adminClient)
+			}
```

Comment:
This looks a bit weird: any reason to set `AdminClient` multiple times within a loop? What if `storageOptFns` is empty and this does not happen, is that OK?

Reply:
Yes,`storageOptFns` is an optional thing. I moved `adminClient` out of this loop.